tracker-edtracker/ftnoir_tracker_edtracker.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ftnoir_tracker_edtracker.h"
+#include "api/plugin-api.hpp"
+#include "compat/math.hpp"
+#include <cinttypes>
+#include <algorithm>
+#include <cmath>
+
+static const int max_value[] = {
+    100,
+    100,
+    100,
+    180,
+    90,
+    180
+};
+
+tracker_edtracker::tracker_edtracker() : pose { 0, 0, 0, 0, 0, 0 } {
+}
+
+tracker_edtracker::~tracker_edtracker() {
+    requestInterruption();
+    wait();
+    if (idevice > 0) close(idevice);
+}
+
+module_status tracker_edtracker::start_tracker(QFrame *) {
+    QString device = QString("/dev/input/js%1").arg(s.dev);
+    idevice = open(device.toUtf8().data(), O_RDONLY | O_NONBLOCK);
+    if (idevice == -1) error("EDTracker not found");
+    notifier = new QSocketNotifier(idevice, QSocketNotifier::Read);
+    notifier->setEnabled(true);
+    QObject::connect(notifier, SIGNAL(activated(int)), this, SLOT(readyRead(int)));
```

Comment:
Don't use Qt signals for this since it happens on the UI thread. Use `poll(2)` and non-blocking reads on the fd.

Reply:
This is even worse since `readyRead()` will run on the UI thread as well.